nanoAOD/plugins/HadTruthProducer.cc

```diff
@@ -73,12 +79,49 @@ HadTruthProducer::produce( edm::Event& iEvent, const edm::EventSetup& iSetup)
     if (nmatched == 2) {
       isHadFrom(trueHad, 6, count, hadFromQuark, hadFromTop);
       matchHad.insert({trueHad.get(), &cand});
-    }
+      matchedGen.push_back(trueHad.get());
+    } else { matchedGen.push_back(nullptr); }
 
     isHadFromTsb.push_back(hadFromQuark);
     isHadFromTop.push_back(hadFromTop);
   }
   
+  for (auto indices : *hadronIndices) {
+    if (abs((*hadronCands)[indices[0]].pdgId()) != 5122) continue;
+    
+    int nmatched = 0;
+    int count=0;
+    int hadFromQuark=-99;
+    bool hadFromTop=false;
+    
+    std::vector<reco::GenParticleRef> momlist;    
+    
+    for (auto i = 1; i <= 2; i++){
```

Comment:
We should be taking the end point from the size of the index list (for if we add other channels here)